test/verify/check-networkmanager-wireguard

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python3 -cimport os, sys; os.execv(os.path.dirname(sys.argv[1]) + "/../common/pywrap", sys.argv)
+
+import subprocess
+import sys
+
+import netlib
+import packagelib
+import testlib
+
+
+class TestWireGuard(packagelib.PackageCase, netlib.NetworkCase):
+    provision = {
+        "machine1": {"address": "192.168.100.11/24", "restrict": False},
+        "machine2": {"address": "192.168.100.12/24", "restrict": False},
+    }
+
+    def testVPN(self):
+        m1 = self.machines["machine1"]
+        m2 = self.machines["machine2"]
+        b = self.browser
+
+        self.login_and_go("/network")
+
+        # Peer 1 (client)
+        m1_port = 51820
+        m1_ip4 = "10.0.0.1"
+        m1_ip6 = "2001::1"
+        b.click("button:contains('Add VPN')")
+        b.wait_visible("#network-wireguard-settings-dialog")
+        iface_name = b.val("#network-wireguard-settings-interface-name-input")
+        b.wait_visible("button:contains('Save'):disabled")
+        if m1.image in ["rhel4edge", "centos-8-stream"] or m1.image.startswith("rhel-8"):
+            b.wait_visible(".pf-v5-c-alert:contains('wireguard-tools package is not installed')")
+            b.click("button:contains('Cancel')")
+            b.wait_not_present("#network-ip-settings-dialog")
+            b.wait_visible("#networking")
+            b.wait_not_present(f"#networking-interfaces th:contains('{iface_name}')")
+            # Skip the rest of the tests for images without wireguard-tools
+            # As without it private/public key, connection over IPv4/IPv6 etc can't be tested
+            return
+
+        # Peer 2 (server)
+        m2_port = 51820
+        m2_ip4 = "10.0.0.2"
+        m2_ip6 = "2001::2"
+        if not m2.ostree_image:
+            m2.execute(f"firewall-cmd --add-port={m2_port}/udp")
+        m2.execute("wg genkey > private")
+        m2_pubkey = m2.execute("wg pubkey < private").strip()
+        m2.execute("ip link add dev wg0 type wireguard")
```

Comment:
I think the Cockpit UI should be able to do all these steps, no?  Let's do that if possible.

Reply:
We're doing in m1 (machine 1). The only purpose of configuring m2 is to just test the connection.

Reply:
We're doing it in m1 (machine 1). The only purpose of configuring m2 is to just test the connection.